src/material/schematics/ng-update/upgrade-rules/hammer-gestures-v9/hammer-template-check.ts

```diff
@@ -8,41 +8,49 @@
 
 import {parse5} from '@angular/cdk/schematics';
 
-/**
- * List of known events which are supported by the "HammerGesturesPlugin" and by
- * the gesture config which was provided by Angular Material.
- */
-const KNOWN_HAMMERJS_EVENTS = [
+/** List of known events which are supported by the "HammerGesturesPlugin". */
+const STANDARD_HAMMERJS_EVENTS = [
   // Events supported by the "HammerGesturesPlugin". See:
   // angular/angular/blob/0119f46d/packages/platform-browser/src/dom/events/hammer_gestures.ts#L19
-  'pan', 'panstart', 'panmove', 'panend', 'pancancel', 'panleft', 'panright', 'panup', 'pandown',
-  'pinch', 'pinchstart', 'pinchmove', 'pinchend', 'pinchcancel', 'pinchin', 'pinchout', 'press',
-  'pressup', 'rotate', 'rotatestart', 'rotatemove', 'rotateend', 'rotatecancel', 'swipe',
-  'swipeleft', 'swiperight', 'swipeup', 'swipedown', 'tap',
-
-  // Events from the Angular Material gesture config.
-  'longpress', 'slide', 'slidestart', 'slideend', 'slideright', 'slideleft'
+  'pan',       'panstart',    'panmove',    'panend',    'pancancel',    'panleft',
+  'panright',  'panup',       'pandown',    'pinch',     'pinchstart',   'pinchmove',
+  'pinchend',  'pinchcancel', 'pinchin',    'pinchout',  'press',        'pressup',
+  'rotate',    'rotatestart', 'rotatemove', 'rotateend', 'rotatecancel', 'swipe',
+  'swipeleft', 'swiperight',  'swipeup',    'swipedown', 'tap',
 ];
 
+/** List of events which are provided by the deprecated Angular Material "GestureConfig". */
+const CUSTOM_MATERIAL_HAMMERJS_EVENS =
+    ['longpress', 'slide', 'slidestart', 'slideend', 'slideright', 'slideleft'];
+
 /**
  * Parses the specified HTML and searches for elements with Angular outputs listening to
  * one of the known HammerJS events. This check naively assumes that the bindings never
  * match on a component output, but only on the Hammer plugin.
  */
-export function isHammerJsUsedInTemplate(html: string): boolean {
+export function isHammerJsUsedInTemplate(html: string):
+    {standardEvents: boolean, customEvents: boolean} {
   const document =
       parse5.parseFragment(html, {sourceCodeLocationInfo: true}) as parse5.DefaultTreeDocument;
-  let result = false;
+  let customEvents = false;
+  let standardEvents = false;
   const visitNodes = nodes => {
     nodes.forEach(node => {
-      if (node.attrs &&
-          node.attrs.some(attr => KNOWN_HAMMERJS_EVENTS.some(e => `(${e})` === attr.name))) {
-        result = true;
-      } else if (node.childNodes) {
+      if (node.attrs) {
+        customEvents = customEvents ||
+            node.attrs.some(
+                attr => CUSTOM_MATERIAL_HAMMERJS_EVENS.some(e => `(${e})` === attr.name));
+        standardEvents = standardEvents ||
+            node.attrs.some(attr => STANDARD_HAMMERJS_EVENTS.some(e => `(${e})` === attr.name));
```

Comment:
This can be most likely improved performance-wise.

Reply:
Should be better now. Surely could be improved much more, but not worth micro-optimizing.